crates/compiler/mono/src/ir.rs

```diff
@@ -4158,6 +4162,42 @@ pub fn with_hole<'a>(
             hole,
         ),
 
+        IngestedFile(bytes, anno) => match &anno.typ {
+            Type::Apply(Symbol::STR_STR, _, _) => Stmt::Let(
+                assigned,
+                Expr::Literal(Literal::Str(
+                    // This is safe because we ensure the utf8 bytes are valid earlier in the compiler pipeline.
+                    arena.alloc(unsafe { std::str::from_utf8_unchecked(&bytes) }.to_owned()),
+                )),
+                Layout::STR,
+                hole,
+            ),
+            Type::Apply(Symbol::LIST_LIST, elem_type, _)
```

Comment:
I think rather than storing the annotation here we should store a `Variable`, then convert that to a `Layout`, and then check whether it's a string or List U8.